split_to_CMYK.jsx

```diff
@@ -1 +1,124 @@
renameLayer.name = "Cyan"

alert(renameLayer)*/



function makeLayers() {
    try {
        if (app.documents.length) {
            var layerColor = ["Cyan","Magenta","Yellow","Black"]
            var doc = activeDocument;
            var layers = doc.artLayers;
            var cmyk = [layerA= layers.add(),
                               layerB= layers.add(),
                               layerC= layers.add(),
                               layerD= layers.add()];

            if (layerColor) {
                cmyk[0].name = layerColor [3];
                cmyk[0].blendMode = BlendMode.MULTIPLY
                cmyk[1].name = layerColor [2];
                cmyk[1].blendMode = BlendMode.MULTIPLY
                cmyk[2].name = layerColor [1];
                cmyk[2].blendMode = BlendMode.MULTIPLY
                cmyk[3].name = layerColor [0];
                cmyk[3].blendMode = BlendMode.MULTIPLY
              
            }
            
            doc.activeLayer = newLayer;
        }
      } catch (e) { 
    }
}
makeLayers ()

/*function splitCyan () {
	var cyan = doc.activeChannels = [doc.channels.getByName('Cyan')];
	app.executeMenuCommand('selectall');
    
}
splitCyan()*/

/*var cyanChannel = doc.activeChannels = [doc.channels.getByName('Cyan')];
var magChannel = doc.activeChannels = [doc.channels.getByName('Magenta')];
var yelChannel = doc.activeChannels = [doc.channels.getByName('Yellow')];
var blackChannel = doc.activeChannels = [doc.channels.getByName('Black')];*/
\ No newline at end of file
+#target photoshop
+#targetengine "main"
+
+var doc = app.activeDocument;
+
+//switch color mode to get 4 channels
+doc.changeMode(ChangeMode.CMYK);
+
+//copies all channels into their own layers and adjusts blending mode
+//reverse order of var layerK - layerC so layers appear in proper order in PS
+function makeLayers() {
+		var layers = doc.artLayers;
+
+		var layerK = layers.add();
+  			layerK.name = 'Black';
+  			layerK.blendMode = BlendMode.NORMAL;
+
+		var layerY = layers.add();
+  			layerY.name = 'Yellow';
+  			layerY.blendMode = BlendMode.MULTIPLY;
+
+		var layerM = layers.add();
+  			layerM.name = 'Magenta';
+  			layerM.blendMode = BlendMode.MULTIPLY;
+
+		var layerC = layers.add();
+  			layerC.name = 'Cyan';
+  			layerC.blendMode = BlendMode.MULTIPLY;
+}
+  //used in splitChannels function
+function selectCopy() {
+        doc.selection.selectAll();
+        doc.selection.copy();
+  }
+//copies channels into corresponding layers
+function splitChannels(targetChannel) {
+    doc.activeLayer = doc.artLayers.getByName("Background");
+    doc.activeChannels = [doc.channels.getByName(targetChannel)];
+    selectCopy();
+    doc.activeLayer = doc.artLayers.getByName(targetChannel);
+    doc.paste();
+}
+
+function addLevelsHueSat(layerName, hue, sat, light) {
+//add levels adj layer
+    doc.activeLayer = doc.artLayers.getByName(layerName);
+    var c2t = function (s) {
+        return app.charIDToTypeID(s);
+    };
```

Comment:
I think you said these were spat out by ScriptListener?  One thing you might want to do is go through these auto-generated functions and clean up the syntax to look more similar to the rest of your code.  For example this line has a `;` after the closing `}`, which isn't a stylistic convention you're doing within your own code throughout the rest of the script.

Matching the syntactical expressions throughout will make the code a little easier to read, and give it a more cohesive feel.